contract/controllers/main.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request
+from odoo.addons.portal.controllers.portal import CustomerPortal,\
+    pager as portal_pager
+
+
+class PortalContract(CustomerPortal):
+
+    def _prepare_portal_layout_values(self):
+        values = super()._prepare_portal_layout_values()
+        model = 'contract.contract'
+        if request.env[model].check_access_rights('read', raise_exception=False):
+            values['contract_count'] = request.env[model].search_count([])
+        else:
+            values['contract_count'] = 0
```

Comment:
For avoiding the coverage drop, you can do (and it's even less lines):
```suggestion
        values['contract_count'] = 0
        if request.env[model].check_access_rights('read', raise_exception=False):
            values['contract_count'] = request.env[model].search_count([])
```